### PR TITLE
[public-api] Publish to NPM

### PR DESCRIPTION
**Pull request by easyCZ**
## Description

Publish public-api typescript defs to NPM so they can be consumed by IDE.

## Related Issue(s)

Fixes #

## Release Notes

```release-note
[public-api] Publish typescript definitions to NPM
```

## Documentation

NONE

### CODE REVIEW

File: components/public-api/typescript/package.json [hunk 1]
```diff
@@ -1,10 +1,8 @@
 {
-  "private": true,
   "name": "@gitpod/public-api",
   "version": "0.1.5",
   "license": "UNLICENSED",
   "files": [
-    "client",
```
Comment (andrew-farries) [inline components/public-api/typescript/package.json:7]: why remove this?

File: components/public-api/typescript/BUILD.yaml [hunk 2]
```diff
@@ -4,13 +4,26 @@ packages:
...
+    env:
+      - DO_PUBLISH=${publishToNPM}
+    argdeps:
+      - npmPublishTrigger
+    deps:
+      - :lib
+      - components/gitpod-protocol:scripts
+    config:
+      commands:
+        - ["node", "components-gitpod-protocol--scripts/publish.js", "${version}", "components-public-api-typescript--lib/package"]
```
Comment (andrew-farries) [inline components/public-api/typescript/BUILD.yaml:29]: How is `node` able to find `publish.js` here (the `components-` don't look like valid paths?)

File: components/public-api/typescript/BUILD.yaml [hunk 3]
```diff
@@ -4,13 +4,26 @@ packages:
...
+      - tsconfig.json
     config:
       packaging: library
       dontTest: true
       commands:
         build: ["yarn", "build"]
       yarnLock: ${coreYarnLockBase}/../yarn.lock
       tsconfig: tsconfig.json
+
+  - name: publish
```
Comment (andrew-farries) [inline components/public-api/typescript/BUILD.yaml:18]: How is this triggered? We already have this:

[mp/public-api-publish-npm](https://github.com/gitpod-io/gitpod/blob/c887240be815c345efbba3fcc2660935637bd767/components/BUILD.yaml#L81-L89)

Does that need to be updated

File: components/public-api/typescript/BUILD.yaml (hunk 2 above)
Comment (easyCZ) [reply components/public-api/typescript/BUILD.yaml:29]: I've no idea! This whole thing looks to be way more complicated than it needs to be but not really well placed to simplify. The code here just mirrors what supervisor-api does

File: components/public-api/typescript/BUILD.yaml (hunk 3 above)
Comment (easyCZ) [reply components/public-api/typescript/BUILD.yaml:18]: Not sure, mirrors supervisor-api setup

**Review by andrew-farries — APPROVED** [review]

**Pull request merged**